packages/kotlin-sdk/sdk/src/main/kotlin/org/dashfoundation/dashsdk/security/WalletStorage.kt

```
@@ -477,21 +745,161 @@ class WalletStorage(
+        unaeProvesRecoverable: Boolean = true,
+        decrypt: () -> ByteArray?,
+    ): Boolean =
+        try {
+            val plain = decrypt()
+            if (plain != null) {
+                plain.fill(0)
+                true
+            } else {
+                false
+            }
+        } catch (e: UserNotAuthenticatedException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)